packages/design-system/src/scalars/components/fragments/textarea-field/textarea-field.tsx

```diff
@@ -84,29 +86,51 @@ const TextareaFieldRaw = React.forwardRef<HTMLTextAreaElement, TextareaProps>(
           })
         : propValue;
 
-    const adjustHeight = (element: HTMLTextAreaElement) => {
-      if (autoExpand) {
+    const adjustHeight = () => {
+      const textarea = document.getElementById(id);
+      if (textarea) {
         // Reset height to allow shrinking
-        element.style.height = "auto";
+        textarea.style.height = "auto";
         // Set to scrollHeight to expand based on content
-        element.style.height = `${element.scrollHeight}px`;
+        textarea.style.height = `${textarea.scrollHeight}px`;
       }
     };
 
     const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
-      if (autoExpand) {
-        adjustHeight(e.currentTarget);
+      // Prevent line breaks if multiline is false
+      let newValue = e.target.value;
+      if (!multiline) {
+        newValue = newValue.replace(/\n/g, "");
       }
 
-      // Transform and directly modify the value before sending it through onChange
-      e.target.value = applyTransformers(e.target.value, {
+      // Transform and directly modify the value
+      newValue = applyTransformers(newValue, {
         lowercase: !!lowercase,
         trim: !!trim,
         uppercase: !!uppercase,
       });
 
+      // Update the textarea value
+      e.target.value = newValue;
+
       // Call the original onChange
       onChange?.(e);
+
+      // Adjust height after processing the value
+      if (autoExpand) {
+        adjustHeight();
+      }
+    };
+
+    const handleKeyDown = (e: React.KeyboardEvent<HTMLTextAreaElement>) => {
+      // Prevent Enter key if multiline is false
+      if (!multiline && e.key === "Enter") {
+        e.preventDefault();
+        e.stopPropagation();
```

Comment:
it should not stop the propagation in case an user wants a custom event. What if the user wants a custom keyDownEvent?

Reply:
Deleted